include/mbgl/map/map.hpp

```diff
@@ -182,10 +183,11 @@ class Map : private util::noncopyable {
     double getDefaultPitch() const;
 
     // Feature queries
-    std::vector<Feature> queryRenderedFeatures(const ScreenCoordinate&, const QueryOptions& options = {});
-    std::vector<Feature> queryRenderedFeatures(const ScreenBox&,        const QueryOptions& options = {});
-    
+    std::vector<Feature> queryRenderedFeatures(const ScreenCoordinate&, const RenderedQueryOptions& options = {});
+    std::vector<Feature> queryRenderedFeatures(const ScreenBox&,        const RenderedQueryOptions& options = {});
+
     AnnotationIDs queryPointAnnotations(const ScreenBox&);
+    std::vector<Feature> querySourceFeatures(const std::string&,        const SourceQueryOptions& options = {});
```

Comment:
GL native exposes an object-oriented API for sources -- move this method to `mbgl::Style::Source`, rename it to `queryFeatures`, and drop the first parameter.